runtime/lua/vim/lsp/util.lua

```diff
@@ -470,6 +470,8 @@ function M.apply_text_document_edit(text_document_edit, index)
   -- `VersionedTextDocumentIdentifier`s version may be null
   --  https://microsoft.github.io/language-server-protocol/specification#versionedTextDocumentIdentifier
   if should_check_version and (text_document.version
+      and text_document.version ~= vim.NIL
```

Comment:
There's a test case that tests for the `vim.NIL` case so I had to add this. Strictly speaking we could probably remove both the test and this given that we normalize `vim.NIL` to `nil` in the rpc layer.

Reply:
I'm ok removing the test and this in the name of simplification.

Reply:
Added a second dedicated commit to remove it, so that the change could be easily reverted individually